### DISCUSSION TITLE
Grouparoo does not run on Windows (Error: spawn npm ENOENT)

### DISCUSSION THREAD

**Discussion by navinkumar0601** [grouparoo/grouparoo#1491] [General] [answered]
I am trying to setup Grouparoo via Node.js and getting following error. I have all known methods, but this is killing me :( 

```
i path: NLlocal
‼ NLlocal\package.json already exists, not modifying
‼ NLlocal\.env already exists, not modifying
‼ NLlocal\.gitignore already exists, not modifying
√ Created directories for config objects.
node:events:355
      throw er; // Unhandled 'error' event
      ^

Error: spawn npm ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:81:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:81:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn npm',
  path: 'npm',
  spawnargs: [ '--version' ]
```

**Comment by evantahler** [comment]
Hi @navinkumar0601, thanks for trying Grouparoo!

Can you share a little more about your environment?
- node version
- operating system
- grouparoo version

Thank you!

**Reply by navinkumar0601** [reply]
Node -v :15.11.0
OS: Windows 10
Grouparoo : Latest version

**Reply by evantahler** [reply]
Ok, thanks!

At the moment, Grouparoo does not run on Windows.  I've opened a PR in which we can track the needed changes over at https://github.com/grouparoo/grouparoo/pull/1493

In the short term, you have 2 options:
* You can use the windows-subsystem-for-linux (WSL) to run Ubuntu-on-windows (https://docs.microsoft.com/en-us/windows/wsl/about)
* You can manually try to make the files `grouparoo init` would have.  You can see the templates [here](https://github.com/grouparoo/grouparoo/tree/main/cli/templates).  I'm a bit skeptical that the `.env` file will work on windows though.

**Comment by bleonard** [comment]
Maybe this? https://stackoverflow.com/questions/43230346/error-spawn-npm-enoent

**Comment by evantahler** [comment, marked as answer]
I am confirming that at this time, Windows is not a supported environment to run Grouparoo.  That said, you can use Windows Subsystem for Linux (WSL) to run Grouparoo within Ubuntu on Windows.  

<img width="1819" alt="Screen Shot 2021-03-18 at 2 51 21 PM" src="https://user-images.githubusercontent.com/303226/111702816-12af8a80-87fa-11eb-8671-26c2bcdbe9d3.png">

The docs have been updated to share this information as well - https://github.com/grouparoo/www.grouparoo.com/pull/328

**Comment by navinkumar0601** [comment]
Got it thanks, now I understand why its failing


T.V. Navin Kumar

________________________________
